tests/framework/results/test_interface.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd
+import pytest
+
+from vivarium.framework.results import ResultsInterface, ResultsManager
+
+
+def _silly_aggregator(_: pd.DataFrame) -> float:
+    return 1.0
+
+
+@pytest.mark.parametrize(
+    (
+        "name, pop_filter, aggregator, requires_columns, requires_values,"
+        " additional_stratifications, excluded_stratifications, when"
+    ),
+    [
+        (
+            "living_person_time",
+            'alive == "alive" and undead == False',
+            _silly_aggregator,
+            None,
+            None,
+            [],
+            [],
+            "collect_metrics",
+        ),
+        (
+            "undead_person_time",
+            "undead == True",
+            _silly_aggregator,
+            None,
+            None,
+            [],
+            [],
+            "time_step__prepare",
+        ),
+    ],
+    ids=["valid_on_collect_metrics", "valid_on_time_step__prepare"],
+)
+def test_register_observation(
+    name,
+    pop_filter,
+    aggregator,
+    requires_columns,
+    requires_values,
+    additional_stratifications,
+    excluded_stratifications,
+    when,
+):
+    mgr = ResultsManager()
+    interface = ResultsInterface(mgr)
+    # interface.set_default_stratifications(["age", "sex"])
+    assert len(interface._manager._results_context._observations) == 0
+    interface.register_observation(
+        name,
+        pop_filter,
+        aggregator,
+        additional_stratifications,
+        excluded_stratifications,
+    )
+    assert len(interface._manager._results_context._observations) == 1
```

Comment:
I'm not that familiar yet w/ this design. This is testing that each of the parameterized tests adds an observation, but do you need to test that they're not overwriting each other? ie register two observations and ensure that the total ._observations == 2?

Reply:
I think it would be good to have a test in which you add two different observations, which I think is what Steve is suggesting here.

Reply:
I'll add this.